lightningd/invoice.c

```diff
@@ -284,22 +281,34 @@ static void json_listinvoice_internal(struct command *cmd,
 				      const jsmntok_t *params,
 				      bool modern)
 {
-	jsmntok_t *label = NULL;
+	jsmntok_t *labeltok = NULL;
+	struct json_escaped *label;
 	struct json_result *response = new_json_result(cmd);
 	struct wallet *wallet = cmd->ld->wallet;
 
 	if (!json_get_params(cmd, buffer, params,
-			     "?label", &label,
+			     "?label", &labeltok,
 			     NULL)) {
 		return;
 	}
 
+	if (labeltok) {
+		label = json_tok_escaped_string(cmd, buffer, labeltok);
+		if (!label) {
```

Comment:
Before this change, we could label an invoice as `null`, `true`, `false`, or a true number, and use it as-is.  This is interesting in the case where `lightning-cli` is used instead of a direct programmatic RPC interface: if you give a number on the command line it is passed as a number.  On previous versions giving `label=0` on the `lightning-cli` command line would succeed in much the same way that giving `label=a` would, but now will `label=0` would fail and would need to add explicit quotes (which would also need to be escaped at the system shell level).  Maybe this is OK, though.

Reply:
Hmm, numeric labels make sense, but we never *really* had them. Label 00 was different from label 0 for example, and fixing that is a bigger job.